### DISCUSSION TITLE
Idea for retroactive sector duration multiplier

### DISCUSSION THREAD

**Discussion by anorth** [filecoin-project/FIPs#476] [Enhancements - Technical]
# Background

FIP-0036 proposed directing a greater share of rewards towards towards sectors with longer up-front committed durations via a multiplier on sector quality computed when a sector is committed or extended. The initial pledge required increases in proportion to that quality multiplier. Along with the projected positive effects on token supply flows, this has the effect of raising the cost of providing storage (as opposed to quality-adjusted power) to the network, which increases with sector duration (longer term loan → higher interest rate), producing diseconomies of timescale. 

There is a legitimate near-term challenge with the current crypto-economic model and environment. A general theme of reducing/delaying net emissions to better balance token supply dynamics seems like a promising direction, and the idea to reward parties who delay emissions makes good sense.

There seems to be an undesirable positive feedback loop in current system design where if net storage onboarding (including expirations) lags, the token supply gets larger, which increases the cost of subsequent onboarding/extension, which would tend to reduce that onboarding.

# Motivation

Direct a greater share of rewards to storage providers that demonstrate long term commitment through their realised sector lifetimes. Improve net onboarding by incentivising extension of existing sectors according to their age.

Dampen the positive feedback loop between pledge and circulating supply: as supply expands, increase the incentive to defer emissions.

# Proposal

Add a duration multiplier to sector quality based on sector age. The longer a sector has been committed, the higher its quality, power, and thus share of reward. The multiplier function is open to design, but I propose a curve from 1.0 at 1 year and approaching an asymptote of some terminal multiplier (arctan?). Such a function will remain applicable (perhaps with a different coefficient) even if the maximum sector duration is greatly increased in the future.

At the point where a multiplier is activated, a sector’s pledge requirement might also increase. I propose that the amount of this increase, as a fraction of the new power, be a network parameter between 0 and 1.0. 

I’m not yet proposing a concrete mechanism for implementing this yet. There are some challenges, such as when and how frequently to activate the multiplier to best approximate a continuously-growing bonus. We should instead reach some consensus about whether this, or other proposals, are widely desirable. Any good mechanism must not disincentivize making long up-front commitments in the first place, however.

# Discussion

A retro-active multiplier introduces an *opportunity cost* to letting a sector expire. A sector that’s about to expire is more profitable than a new one. A provider is thus incentivised to maintain their current storage commitments even as returns fluctuate within some range, since they can’t get back to the same rate of return if they drop and then seal new sectors. This contrasts with the lack of significant incentive to extend sectors today.

The impact on net onboarding and emissions is thus to incentivise sector extension. This would provide a stabilising incentive on sector commitments, and reward the deferral of the eventual release of that sector’s pledge.

### Pledge increase parameter

Whether or how much to increase a sector’s pledge when claiming a multiplier affects the relative profitability of extension under different conditions. Modulo changes in network total power:

- If a sector’s pledge requirement is increased to match the new power and token supply, like extension today, then the new return-on-pledge will be equal to that of a new sector. The return on fixed costs (i.e. hardware) improves. This matches current network behaviour.
    - If the token supply has increased since initial commitment, this will tend to reduce a sector’s profitability, as pledge increases more than reward, reducing the realised multiplier. If the token supply has decreased, the sector’s profitability improves by the multiplier (the high-water-mark pledge requirement is maintained).
- If a sector’s pledge requirement is held constant, or increased by some fraction < 1.0 of the amount to match a new sector, the return-on-pledge for extending will be *better* than that of new sector.
    - As the token supply increases, the profitability of, and hence opportunity cost of dropping, an extended sector increases as reward grows faster than pledge. If the token supply decreases, return-on-pledge improves by the multiplier amount as above.

With a pledge increase rate <1.0, the dynamic of increasing the incentive to maintain sectors as the token supply, and thus new pledge requirement, grows would have a damping effect on the supply→profitability feedback loop. As token supply grows, the relative return on existing sectors grows too, incentivising their maintenance and thus deferring their pledge release and acting against token supply growth.

Unlike today, providers are also incentivised to extend sectors even when pledge requirements have fallen. The incentive to drop sectors and re-seal with lower pledge is only attractive after the marginal returns of that approach exceed the multiplier awarded to maintaining the sector.

Depending on the multiplier parameters, as sectors age the “carrot” of future additional revenue could well exceed the direct cost “stick” of termination fees, which is the only existing incentive against early termination of sectors.

### Impact on storage commitments

In recent history, net raw byte storage capacity growth is close to zero, with expirations matching pace with new sector commitments (QA power growth is higher from verified deal onboarding). At the upper bound, if expirations reduced from their current rate to zero, the net raw byte growth rate would approach the gross onboarding rate of ~12PiB / day.

### Impact on token supply

Retroactive multipliers would have a lesser impact on token flows than that aimed for by FIP-0036. At the upper bound, if sector expirations reduced from their current rate to zero, the reduction in released pledge (~150K daily) would account for about 30% of the current net supply increase (~500K daily). 

However, retro-active multipliers don’t face the tricky problem of inducing additional investment while reducing sector returns, and the risks of the associated feedback loop.

### Applicability to existing sectors

Retro-active multipliers should be applicable to both already-committed sectors and newly committed ones. A mechanism must be designed to facilitate the orderly activation of multipliers across the existing commitment base without causing a shock to either total QA power or any individual provider’s share of it. A good multiplier activation function should leave the storage provider relatively ambivalent as to the timing of claiming it, up to a month or two.

### Increased rewards to existing providers

Directing a greater share of rewards to older sectors has the effect of increasing the relative profitability and share of consensus power of existing providers as compared to new entrants. It rewards demonstrated commitment rather than projected commitment. This would increase the relative advantage to the longest serving existing storage providers. An inflexion point and asymptote on the multiplier function would limit that advantage to some fixed amount.

This is not necessarily desirable, and worth robust discussion.

### Alternatives

This proposal is presented as one option among multiple approaches to improving network economics. 

Retro-active multipliers need not be considered mutually exclusive with up-front commitment duration multipliers of the style proposed by FIP-0036. These could work together to incentivise both longer up-front commitments and extension of existing ones. 

This discussion should not indicate commitment to one approach: we should decide as a community which of these or other ideas is most advantageous.

**Comment by geoff-vball** [comment]
Thanks for posting this discussion! Any way of reducing the current feedback loop is worth investigation. I'll need some more time to think on everything here and would love to hear others thoughts on this.

**Comment by AxCortesCubero** [comment]
Thanks for continuing the post FIP 36 discussion, this is an interesting alternative! 

My first thought is that it is less clear to me why having a sector that has existed for a long time would be valuable for the network, and should be rewarded more.

The way I understood the duration multiplier is that having a long term commitment adds value to the sector. We have now a guarantee that this sector will stay in place for longer, guarantee of future data stability as well as the value of knowing collateral will be locked for longer, which is good for controlling the circulating supply. From the SP side, making a long term commitment also involves a higher risk, so it makes sense to require further compensation.

So the value comes from the guarantee of what will happen in he future, this is useful and valuable and should be rewarded.

So I dont understand what is the source of value of a sector that was long and renewed in the past, why should it be rewarded? For instance the risk factor isn't it, that is completely taken out of the equation now.

**Reply by anorth** [reply]
It's a different perspective than rewards for up-front commitments. In this proposal's framing, a sector that has existed for a long time _was already_ valuable for the network, and is rewarded after demonstrating that value. 

The expectation of this reward creates the incentive to commit sectors for a long period, not unlike a commitment duration multiplier. But the actual payment of reward is deferred (rather than kicking in immediately at commitment) and also conditional upon and scales with _continued_ commitment.

**Reply by anorth** [reply]
> We have now a guarantee that this sector will stay in place for longer

I think we should be careful with statements like this. We have no such guarantee, we just know the value of a penalty which would be paid if the commitment is broken. With retro-active multipliers, the penalty is the opportunity cost of a lost multiplier that _would have been paid_ if the sector was retained.

> the value of knowing collateral will be locked for longer

I think that actually locking the collateral is what's valuable. I don't have a good sense for the concrete nature of any additional value to the network of "knowing" (up to some penalty) ahead of time that the collateral was going to be locked. It may be non-zero, but can we quantify it as being non-negligible?

**Comment by AxCortesCubero** [comment]
> As token supply grows, the relative return on existing sectors grows too, incentivising their maintenance and thus deferring their pledge release and acting against token supply growth.

This part is also not super clear to me what you mean. If token supply grows, not sure what this has to do with their relative return on existing sectors, or what is the definition of this. But also pledge generally would also be increasing if supply is going up, so there would already exist an incentive to extend the sector, rather than let expire and seal a new sector.

**Reply by anorth** [reply]
By relative return, I mean that an existing sector will enjoy a higher return on pledge than a new sector.

> there would already exist an incentive to extend the sector

There isn't, because pledge is currently re-calculated when extending a sector. Extending a sector and committing a new one currently result in equivalent return-on-pledge profiles (or worse for the extension, because the recalculation has a floor at the already-pledged amount).

So simply not recalculating pledge upwards when extending is a change to the current protocol, and possibly a worthwhile one.
